maplibre_gl_example/lib/map_snapshot.dart

```diff
@@ -0,0 +1,307 @@
+        style: TextStyle(
+          fontSize: 48 * pixelRatio,
+        ),
+      ),
+      textDirection: TextDirection.ltr,
+    );
+    textPainter.layout();
+    textPainter.paint(
+      canvas,
+      Offset(size / 2 - textPainter.width / 2, size / 2 - textPainter.height / 2),
+    );
+    
+    final picture = recorder.endRecording();
+    final image = await picture.toImage(size.toInt(), size.toInt());
+    final byteData = await image.toByteData(format: ui.ImageByteFormat.png);
+    
+    return byteData!.buffer.asUint8List();
+  }
+
+  Future<Uint8List> _createFrenchFlagIcon([double pixelRatio = 1.0]) async {
+    final recorder = ui.PictureRecorder();
+    final canvas = Canvas(recorder);
+    final size = 64.0 * pixelRatio;
+    final flagWidth = 51.2 * pixelRatio;
+    final flagHeight = 38.4 * pixelRatio;
+    final stripeWidth = 17.066666666666666 * pixelRatio;
+    
+    // Calculate flag position (centered)
+    final flagX = (size - flagWidth) / 2;
+    final flagY = (size - flagHeight) / 2;
+    
+    // Draw French flag (blue, white, red vertical stripes)
+    // Blue stripe
+    final bluePaint = Paint()..color = const Color(0xFF0055A4);
+    canvas.drawRect(Rect.fromLTWH(flagX, flagY, stripeWidth, flagHeight), bluePaint);
+    
+    // White stripe
+    final whitePaint = Paint()..color = Colors.white;
+    canvas.drawRect(Rect.fromLTWH(flagX + stripeWidth, flagY, stripeWidth, flagHeight), whitePaint);
+    
+    // Red stripe
+    final redPaint = Paint()..color = const Color(0xFFEF4135);
+    canvas.drawRect(Rect.fromLTWH(flagX + stripeWidth * 2, flagY, stripeWidth, flagHeight), redPaint);
+    
+    // Draw border
+    final borderPaint = Paint()
+      ..color = Colors.black
+      ..style = PaintingStyle.stroke
+      ..strokeWidth = 1 * pixelRatio;
+    canvas.drawRect(Rect.fromLTWH(flagX, flagY, flagWidth, flagHeight), borderPaint);
+    
+    final picture = recorder.endRecording();
+    final image = await picture.toImage(size.toInt(), size.toInt());
+    final byteData = await image.toByteData(format: ui.ImageByteFormat.png);
+    
+    return byteData!.buffer.asUint8List();
+  }
+
+  Future<void> _takeSnapshot() async {
+    setState(() {
+      _isLoading = true;
+      _error = null;
+      _snapshotImage = null;
+      _currentCity = null;
+    });
+
+    try {
+      // Define random camera positions for different cities
+      final cities = [
+        {'name': 'Beijing', 'lat': 39.9042, 'lng': 116.4074},
+        {'name': 'San Francisco', 'lat': 37.7749, 'lng': -122.4194},
+        {'name': 'London', 'lat': 51.5074, 'lng': -0.1278},
+        {'name': 'Sydney', 'lat': -33.8688, 'lng': 151.2093},
+      ];
+      
+      // Pick a random city
+      final randomCity = cities[DateTime.now().millisecond % cities.length];
+      
+      final cameraPosition = CameraPosition(
+        target: LatLng(randomCity['lat'] as double, randomCity['lng'] as double),
+        zoom: _currentZoom,
+      );
```

Comment:
Adjust the casting type of the variables
```suggestion
      final cameraPosition = CameraPosition(
        target: LatLng(randomCity['lat']! as double, randomCity['lng']! as double),
        zoom: _currentZoom,
      );
```